CIP-XXXX/README.md

```diff
@@ -0,0 +1,1597 @@
+
+Only the portions of the witness set that were signed as a result of this call are returned to encourage dApps to verify the contents returned by this endpoint while building the final transaction.
+
+
+##### SignData
+
+```
+{
+  "operation": "signData",
+  "request": {
+    "type": "object",
+    "properties": {
+      "addr": { "$ref": "#/cip-116/Address" },
+      "payload": { "$ref": "#/cip-116/ByteString" }
+    },
+    "required": ["addr", "payload"]
+  },
+  "response": { "$ref": "#/appendix/DataSignature" },
+  "errors": [
+    { "$ref": "#/appendix/APIError" },
+    { "$ref": "#/appendix/DataSignError" }
+  ]
+}
+```
+
+*[Is bytestring the correct type here?]*
+
+This endpoint utilizes the [CIP-0008 signing spec](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0008/README.md) for standardization/safety reasons. It allows the dApp to request the user to sign a payload conforming to said spec. The user's consent MUST be requested and the message to sign shown to the user. The payment key from `addr` will be used for base, enterprise and pointer addresses to determine the EdDSA25519 key used. The staking key will be used for reward addresses. This key will be used to sign the `COSE_Sign1`'s `Sig_structure` with the following headers set:
+
+- `alg` (1) - must be set to EdDSA (-8)
+- `kid` (4) - Optional, if present must be set to the same value as in the COSE_key specified below. It is recommended to be set to the same value as in the "address" header.
+- `address` - must be set to the raw binary bytes of the address as per the binary spec
+
+The payload is not hashed and no `external_aad` is used.
+
+If the payment key for `addr` is not a P2Pk address then `DataSignError` will be returned with error code `AddressNotPK`. `ProofGeneration` error code shall be returned if the wallet cannot generate a signature (i.e. the wallet does not own the requested payment private key), and `UserDeclined` will be returned if the user refuses the request. The return shall be a `DataSignature` with `signature` set to the hex-encoded bytes of the `COSE_Sign1` object specified above and `key` shall be the hex-encoded bytes of a `COSE_Key` structure with the following headers set:
+
+- `kty` (1) - must be set to `OKP` (1)
+- `kid` (2) - Optional, if present must be set to the same value as in the `COSE_Sign1` specified above.
+- `alg` (3) - must be set to `EdDSA` (-8)
+- `crv` (-1) - must be set to `Ed25519` (6)
+- `x` (-2) - must be set to the public key bytes of the key used to sign the `Sig_structure`
+
+*[What are these numbers appearing next to params? Copied them from cip-30 but not sure]*
```

Comment:
> *[What are these numbers appearing next to params? Copied them from cip-30 but not sure]*

I haven't had a chance to read up on it but this seems to have further explanation: https://www.rfc-editor.org/rfc/rfc9052.html#name-cose-key-common-parameters